include/oneapi/dpl/pstl/iterator_impl.h

```diff
@@ -80,17 +80,17 @@ template <typename... _Types>
 class zip_forward_iterator
 {
     static const ::std::size_t __num_types = sizeof...(_Types);
-    typedef typename ::std::tuple<_Types...> __it_types;
+    typedef typename oneapi::dpl::__internal::tuple<_Types...> __it_types;
 
   public:
     typedef ::std::make_signed_t<::std::size_t> difference_type;
-    typedef ::std::tuple<typename ::std::iterator_traits<_Types>::value_type...> value_type;
-    typedef ::std::tuple<typename ::std::iterator_traits<_Types>::reference...> reference;
-    typedef ::std::tuple<typename ::std::iterator_traits<_Types>::pointer...> pointer;
+    typedef oneapi::dpl::__internal::tuple<typename ::std::iterator_traits<_Types>::value_type...> value_type;
+    typedef oneapi::dpl::__internal::tuple<typename ::std::iterator_traits<_Types>::reference...> reference;
+    typedef oneapi::dpl::__internal::tuple<typename ::std::iterator_traits<_Types>::pointer...> pointer;
```

Comment:
This isn't an argument against these changes, but to provide some context / history:

In #1432 (still unmerged), we removed a similar change because `zip_forward_iterator` was only used on the host side, not in hetero / dpcpp backend, and therefore did not need the tuple to be trivially copyable.